File: notifications/src/main/java/io/castled/notifications/logger/CastledLogger.java
package io.castled.notifications.logger;

import android.util.Log;

import androidx.annotation.NonNull;

import io.castled.notifications.Logger;

public class CastledLogger implements Logger {

    private final String tag;

    private CastledLogger(String tag) {
        this.tag = tag;
    }

    @Override
    public void verbose(@NonNull String message) {
        Log.v(tag, message);
    }

    @Override
    public void debug(@NonNull String message) {
        Log.d(tag, message);
    }

    @Override
    public void info(@NonNull String message) {
        Log.i(tag, message);
    }

    @Override
    public void warning(@NonNull String message) {
        Log.w(tag, message);
    }

    @Override
    public void error(@NonNull String message) {
        Log.e(tag, message);
    }

    @Override
    public void error(@NonNull String message, @NonNull Throwable throwable) {
        Log.v(tag, message, throwable);
    }

    public static CastledLogger getInstance(String tag) {
        return new CastledLogger(tag);
    }
}

File: notifications/src/main/java/io/castled/notifications/logger/CastledLogger.kt
package io.castled.notifications.logger

import android.util.Log

internal class CastledLogger private constructor(private val tag: String) : Logger {
    override fun verbose(message: String) {
        Log.v(tag, message)
    }

    override fun debug(message: String) {
        Log.d(tag, message)
    }

    override fun info(message: String) {
        Log.i(tag, message)
    }

    override fun warning(message: String) {
        Log.w(tag, message)
    }

    override fun error(message: String) {
        Log.e(tag, message)
    }

    override fun error(message: String, throwable: Throwable) {
        Log.e(tag, message, throwable)
    }

    companion object {
        fun getInstance(tag: String): CastledLogger {
            return CastledLogger(tag)
        }
    }
}